pkgs/development/tools/analysis/hopper/default.nix

```diff
@@ -1,42 +1,57 @@
-{ stdenv, fetchurl, pkgs, lib }:
+{
+  stdenv, fetchurl, lib,
 
-stdenv.mkDerivation rec {
+  autoPatchelfHook,
+
+  wrapQtAppsHook,
+
+  libbsd,
+  python27,
+  gmpxx,
+}:
+let
+
+in stdenv.mkDerivation rec {
   pname    = "hopper";
-  version = "4.5.7";
+  version = "4.5.19";
   rev = "v${lib.versions.major version}";
 
   src = fetchurl {
     url = "https://d2ap6ypl1xbe4k.cloudfront.net/Hopper-${rev}-${version}-Linux.pkg.tar.xz";
-    sha256 = "1ce7a0f13126a940398aa8da4a74e250dff0401074f30446a8840ac3dbb902c0";
+    sha256 = "1c9wbjwz5xn0skz2a1wpxyx78hhrm8vcbpzagsg4wwnyblap59db";
   };
 
   sourceRoot = ".";
 
-  ldLibraryPath = with pkgs; stdenv.lib.makeLibraryPath  [
-libbsd.out libffi.out gmpxx.out python27Full.out python27Packages.libxml2 qt5.qtbase zlib  xlibs.libX11.out xorg_sys_opengl.out xlibs.libXrender.out gcc-unwrapped.lib
+  nativeBuildInputs = [
+    wrapQtAppsHook
+    autoPatchelfHook
   ];
 
-  nativeBuildInputs = [ pkgs.qt5.wrapQtAppsHook ];
-
-  qtWrapperArgs = [ ''--suffix LD_LIBRARY_PATH : ${ldLibraryPath}'' ];
+  buildInputs = [
+    libbsd
+    python27
+    gmpxx
+  ];
 
   installPhase = ''
     mkdir -p $out/bin
     mkdir -p $out/lib
     mkdir -p $out/share
+
     cp $sourceRoot/opt/hopper-${rev}/bin/Hopper $out/bin/hopper
     cp -r $sourceRoot/opt/hopper-${rev}/lib $out
```

Comment:
I do not like all these libraries being vendored, could you perhaps only copy those that are not available in nixpkgs and add the rest to `buildInputs`?

```
── lib
│  ├── libBlocksRuntime.so.0
│  ├── libdispatch.so -> libdispatch.so.1
│  ├── libdispatch.so.0.1.3.1
│  ├── libdispatch.so.1 -> libdispatch.so.0.1.3.1
│  ├── libform.so.5
│  ├── libgnustep-base.so -> libgnustep-base.so.1.24
│  ├── libgnustep-base.so.1.24 -> libgnustep-base.so.1.24.9
│  ├── libgnustep-base.so.1.24.9
│  ├── libHopperCore.so
│  ├── libkqueue.so.0
│  ├── libncurses.so.5
│  ├── libobjc.so -> libobjc.so.4.6
│  ├── libobjc.so.4.6
│  ├── libobjcxx.so -> libobjcxx.so.4.6
│  ├── libobjcxx.so.4.6
│  ├── libpthread_workqueue.so.0
│  └── libtinfo.so.5
```

Reply:
* libBlocksRuntime.so.0
  - [ ] copy
* libdispatch.so
  - [ ] copy
* libdispatch.so.0.1.3.1
  - [ ] copy
* libdispatch.so.1
  - [ ] copy
* libform.so.5
  - [ ] copy
  - [ ] ncurses5.out
  - [ ] (gcc-arm-embedded-4_9.out)
  - [ ] (androidStudioPackages.dev.out)
  - [ ] (androidStudioPackages.canary.out)
  - [ ] (android-studio.out)
  - [ ] (androidStudioPackages.beta.out)
* libgnustep-base.so
  - [ ] copy
  - [ ] gnustep.base.out
* libgnustep-base.so.1.24
  - [ ] copy
* libgnustep-base.so.1.24.9
  - [ ] copy
* libHopperCore.so
  - [ ] copy
* libkqueue.so.0
  - [ ] copy
* libncurses.so.5
  - [ ] copy
  - [ ] scilab-bin.out
  - [ ] ncurses5.out
  - [ ] (gcc-arm-embedded-4_9.out)
  - [ ] (androidStudioPackages.dev.out)
  - [ ] (androidStudioPackages.canary.out)
  - [ ] (android-studio.out)
  - [ ] (androidStudioPackages.beta.out)
* libobjc.so
  - [ ] copy
  - [ ] gnustep.libobjc.out
* libobjc.so.4.6
  - [ ] copy
  - [ ] gnustep.libobjc.out
* libobjcxx.so
  - [ ] copy
* libobjcxx.so.4.6
  - [ ] copy
* libpthread_workqueue.so.0
  - [ ] copy
* libtinfo.so.5
  - [ ] copy
  - [ ] tlf.out
  - [ ] ncurses5.out
  - [ ] (gcc-arm-embedded-4_9.out)
  - [ ] jetbrains.clion.out
  - [ ] arduino-core.out
  - [ ] arduino.out
  - [ ] (androidStudioPackages.dev.out)
  - [ ] (androidStudioPackages.canary.out)
  - [ ] (android-studio.out)
  - [ ] (androidStudioPackages.beta.out)

generated with:
```
for l in *; do [ ! -f "${l}" ] && continue; echo "* ${l}"; echo "  - [ ] copy"; nix-locate -1 -w "${l}" | sed -e 's/^/  - [ ] /' ; done | xclip
```

Reply:
`libpthread_workqueue.so.0` might come from https://github.com/mheily/libpwq

Reply:
* libBlocksRuntime.so.0
  - [x] copy
* libdispatch.so.0.1.3.1
  - [x] copy
* libform.so.5
  - [ ] copy
  - [x] ncurses5.out
  - [ ] (gcc-arm-embedded-4_9.out)
  - [ ] (androidStudioPackages.dev.out)
  - [ ] (androidStudioPackages.canary.out)
  - [ ] (android-studio.out)
  - [ ] (androidStudioPackages.beta.out)
* libgnustep-base.so.1.24.9
  - [x] copy
* libHopperCore.so
  - [x] copy
* libkqueue.so.0
  - [x] copy
* libncurses.so.5
  - [ ] copy
  - [ ] scilab-bin.out
  - [x] ncurses5.out
  - [ ] (gcc-arm-embedded-4_9.out)
  - [ ] (androidStudioPackages.dev.out)
  - [ ] (androidStudioPackages.canary.out)
  - [ ] (android-studio.out)
  - [ ] (androidStudioPackages.beta.out)
* libobjc.so.4.6
  - [ ] copy
  - [x] gnustep.libobjc.out
* libobjcxx.so.4.6
  - [ ] copy
* libpthread_workqueue.so.0
  - [ ] copy
* libtinfo.so.5
  - [ ] copy
  - [ ] tlf.out
  - [x] ncurses5.out
  - [ ] (gcc-arm-embedded-4_9.out)
  - [ ] jetbrains.clion.out
  - [ ] arduino-core.out
  - [ ] arduino.out
  - [ ] (androidStudioPackages.dev.out)
  - [ ] (androidStudioPackages.canary.out)
  - [ ] (android-studio.out)
  - [ ] (androidStudioPackages.beta.out)


generated with:

```
for l in *; do [ -L "${l}" ] && continue; echo "* ${l}"; echo "  - [ ] copy"; nix-locate -1 -w "${l}" | sed -e 's/^/  - [ ] /' ; done | xclip
```

Reply:
done